src/pages/index.js

```diff
@@ -13,111 +14,287 @@ import {
-  popupSelector: cardModalSelector,
+  modalSelector: cardModalSelector,
   handleFormSubmit: (inputValues) => {
-    const name = inputValues.title;
-    const link = inputValues.url;
+    modalFormImage.renderLoading(true);
+    // console.log('inputValues: ', inputValues)
 
-    renderCard({ link, name }, cardList);
+    api
+      .addCard(inputValues)
+      .then((inputValues) => {
+        renderCard(inputValues);
+        modalFormImage.close();
+      })
+      .catch((err) => {
+        console.error(err);
+  })
+      .finally(() => {
+        modalFormImage.renderLoading(false);
+        // if ("add-card-form" in formValidators) {
+        //   formValidators["add-card-form"].resetValidation();
+        // }
+      });
   },
+  
+  loadingText: "Saving...",
 });
 
-const cardSection = new Section(
-  {
-    data: initialCards,
-    renderer: renderCard,
+const deleteModal = new PopupWithConfirmation({
+  handleFormSubmit: () => {
+    deleteModal._handleFormSubmit();
   },
-  cardListSelector
-);
-cardSection.renderItems();
+  modalSelector: cardModalDelete,
+  loadingText: "Deleting...",
+});
+
+modalFormImage.setEventListeners();
+modalWithImage.setEventListeners();
+modalFormUser.setEventListeners();
+modalWithFormUser.setEventListeners();
+deleteModal.setEventListeners();
+
+
 
-editFormValidator.enableValidation();
-addFormValidator.enableValidation();
+
+/**************
+ * VALIDATION *
+ **************/
+// const addFormValidator = new FormValidator(validationSettings, addCardEditForm);
+// addFormValidator.enableValidation();
+
+// const editFormValidator = new FormValidator(
+//   validationSettings,
+//   profileEditForm
+// );
+
+// editFormValidator.enableValidation();
+
+/*********************
+ * ENABLE VALIDATION *
+ *********************/
+const formValidators = {};
+
+const enableValidation = (validationSettings) => {
+  const formList = Array.from(
+    document.querySelectorAll(validationSettings.formSelector)
+  );
+  formList.forEach((formElement) => {
+    const validator = new FormValidator(validationSettings, formElement);
+
+    const formId = formElement.getAttribute("id");
+
+    formValidators[formId] = validator;
+    validator.enableValidation();
+  });
+};
+
+enableValidation(validationSettings);
 
 /*******************
  * EVENT LISTENERS *
  *******************/
 
 profileEditButton.addEventListener("click", () => {
-  modalWithFormUser.open();
+  modalFormUser.open();
   const userData = userInfo.getUserInfo();
   modalNameInput.value = userData.userName;
   modalDescriptionInput.value = userData.userDescription;
-  editFormValidator.resetValidation();
+  if ("edit-modal-form" in formValidators) {
+    formValidators["edit-modal-form"].resetValidation();
+  }
 });
 addNewCardButton.addEventListener("click", () => {
-  addFormValidator.resetValidation();
   modalFormImage.open();
+  if ("add-card-form" in formValidators) {
+    formValidators["add-card-form"].resetValidation();
```

Comment:
Seems like this is not working, the button is enabled when I open the modal, but it should be disabled

Reply:
Don't forget to fix this. The button should be disabled when you open the card form.